File: Webpage/server/webpage.server/webpage.server/src/main/java/abbruzzese/webpage/webpage/server/webpage/Mappers/NewsletterMapper.java
package abbruzzese.webpage.webpage.server.webpage.Mappers;

import org.springframework.stereotype.Component;
import abbruzzese.webpage.webpage.server.webpage.DTO.DTONewsletter;
import abbruzzese.webpage.webpage.server.webpage.entities.Newsletter;

@Component(value = "NewsletterMapper")
public class NewsletterMapper {

    public DTONewsletter NewslettertoDTO(Newsletter newsletter){
        DTONewsletter dto=new DTONewsletter();
        dto.setEmail(newsletter.getEmail());
        return dto;
    }

    public Newsletter DTOtoNewsletter(DTONewsletter dto){
        Newsletter newsletter= new Newsletter();
        newsletter.setEmail(dto.getEmail());
        return newsletter;
    }

}

File: PHP_API/MAPPER/NewsletterMapper.php
<?php

namespace App\Mapper;

use App\Entity\Newsletter;
use App\DTO\DTONewsletter;

class NewsletterMapper {

    public static function NewslettertoDTO(Newsletter $newsletter) {
       
        return new DTONewsletter(
            $newsletter->getEmail(),
        );
    }

    public static function DTOtoNewsletter(DTONewsletter $newsletterdto) {
        return new Newsletter(
            $newsletterdto->getEmail(),
        );
    }
}

?>